File: lib/scanner_callback.dart
import 'package:honeywell_scanner/scanned_data.dart';

typedef OnScannerDecodeCallback = Function(ScannedData? scannedData);
typedef OnScannerErrorCallback = Function(Exception error);

/// Callback of the decoding process
abstract class ScannerCallback {
  /// Called when decoder has successfully decoded the code
  ///
  /// @param scannedData Encapsulates the result of decoding a barcode within an image
  void onDecoded(ScannedData? scannedData);

  /// Called when error has occurred
  ///
  /// @param error Exception that has been thrown
  void onError(Exception error);
}

File: android/src/main/java/com/plugin/flutter/honeywell_scanner/ScannerCallBack.java
package com.plugin.flutter.honeywell_scanner;

import android.app.Activity;
import android.os.Handler;

/**
 * Created by krrigan on 5/31/19.
 */

public interface ScannerCallBack
{
    /**
     * Called when decoder has successfully decoded the code
     * <br>
     * Note that this method always called on a worker thread
     *
     * @param scannedData Encapsulates the result of decoding a barcode within an image
     * @see Handler
     * @see Activity#runOnUiThread(Runnable)
     */
    void onDecoded(ScannedData scannedData);
    /**
     * Called when error has occurred
     * <br>
     * Note that this method always called on a worker thread
     *
     * @param error Exception that has been thrown
     * @see Handler
     * @see Activity#runOnUiThread(Runnable)
     */
    void onError(Exception error);
}